Cpp_test/lib/ipc-queue-receive.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ipc-queue-receive.h"
+#include <ctime>
+#include <iostream>
+
+ipcQueueReceive::~ipcQueueReceive()
+{
+	mq_close(this->mqd);
+	mq_unlink(this->info.method_name);
+}
+
+void ipcQueueReceive::init()
+{
+
+	std::cout << "Waiting for sender" << std::endl;
+	do
+	{
+		this->mqd = mq_open(this->info.method_name, O_RDONLY | O_NONBLOCK, 0660, &(this->attr));
+	} while (this->mqd == -1);
+
+	if (this->mqd == -1)
+		throw std::runtime_error("ERROR: Fail to open receive queue. ");
+	else
+		std::cout << "/dev/mqueue" << this->info.method_name << " is opened." << std::endl;
+
+	this->open_file();
+}
+
+void ipcQueueReceive::transfer()
+{
+	std::vector<char> buffer(this->attr.mq_msgsize);
+	long read_bytes{0};
+	std::cout << "Waiting for new file..." << std::endl;
+	errno = 0; // clear error number
+	while (errno != EAGAIN )
```

Comment:
So if there is another error in the mq_receive, this loop will continue indefinitively.

Reply:
Did you think what happened if ipc_receive can go in this while before ipc_send can add a message?

Reply:
> So if there is another error in the mq_receive, this loop will continue indefinitively.

I forgot about this problem; I will use the timeout with mq_timedreceive() to handle this problem